src/components/ha-icon-picker.ts

```diff
@@ -58,7 +58,7 @@ export class HaIconPicker extends LitElement {
         item-label-path="icon"
         .value=${this._value}
         allow-custom-value
-        .filteredItems=${iconItems}
+        .filteredItems=${this._filterIcons(this._filterString)}
```

Comment:
```suggestion
        .filteredItems=${this._filterIcons(this._filterString, ICONS)}
```
Otherwise, the items are not updated when the icons are loaded.

Reply:
Harmless change, but I don't understand why that makes a difference?

Reply:
`_filterIcons` is a memoized function. Once icons are loaded, `requestUpdate` is call but the parameters of `_filterIcons` are the same because `ICONS` are not passed as parameter.

Reply:
So the default for the parameter won't be checked by `memoizeOne`?

Reply:
Hmm... just tested.  I guess not.  Good to know.

Reply:
No it only checks parameters passed to the function.